### PR TITLE
[ENH] Add `fill_value` and `explicit` to the `complete` function

### LINKED ISSUE #1019: [ENH] Add `fill_value` and `explicit` parameters to `complete`
# Brief Description

fill null values generated during `complete` and decide if existing nulls should be filled, or only nulls generated from the `complete` function. idea inspired by the updates to the [complete](https://tidyr.tidyverse.org/reference/complete.html) function in tidyr.

# Example API

```python
# source dataframe
df =  dict(
  group = (1,2, 1, 2),
item_id = (1,2, 2, 3),
 item_name = ("a", "a", "b", "b"),
value1 = (1, pd.NA, 3, 4),
 value2 = range(4,8)
 )

df  = pd.DataFrame(df)

   group  item_id item_name value1  value2
0      1        1         a      1       4
1      2        2         a   <NA>       5
2      1        2         b      3       6
3      2        3         b      4       7

# fill value and explicit set to True
df.complete('group', ('item_id', 'item_name'), fill_value = 0, sort = True, explicit = True) 

   group  item_id item_name  value1  value2
0      1        1         a       1     4
1      1        2         a       0     0
2      1        2         b       3     6
3      1        3         b       0     0
4      2        1         a       0     0
5      2        2         a       0     5
6      2        2         b       0     0
7      2        3         b       4     7

df.complete('group', ('item_id', 'item_name'), fill_value = 0, sort = True, explicit = False) 

   group  item_id item_name  value1  value2
0      1        1         a       1     4
1      1        2         a       0     0
2      1        2         b       3     6
3      1        3         b       0     0
4      2        1         a       0     0
5      2        2         a    <NA>     5
6      2        2         b       0     0
7      2        3         b       4     7
``` 

It is easy to do a `fillna` after `complete`; however it looks cumbersome filling only the nulls generated from the `complete` function. I'll gladly take suggestions for filling only the null values without adding new parameters.

### PR DESCRIPTION
**Pull request by samukweku**
# PR Description

Please describe the changes proposed in the pull request:

- Add `fill_value` and `explicit` parameters
- `explicit` determines if only nulls introduced via `complete` will be filled, or all nulls exisitng in the dataframe.

**This PR resolves #1019 .**

# PR Checklist

Please ensure that you have done the following:

1. [ ] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [ ] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [ ] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

**Comment by ericmjl** [comment]
🚀 Deployed on https://deploy-preview-1030--pyjanitor.netlify.app

File: janitor/functions/complete.py [hunk 1]
```diff
@@ -14,24 +14,26 @@
 def complete(
     df: pd.DataFrame,
     *columns,
-    sort: bool = False,
+    sort: bool = True,
```
Comment (thatlittleboy) [inline janitor/functions/complete.py]: 1. Modifying the default args is a change worth documenting explicitly in the Changelog.
2. Could you also include this change into the PR description and give some rationale for this change? Btw, Not saying we should or should not change the default arg of `sort`, just that it is important to note down the reason for doing so for future reference.

File: janitor/functions/complete.py [hunk 2]
```diff
@@ -85,14 +86,72 @@ def complete(
...
-    :param sort: Sort DataFrame based on *columns. Default is `False`.
+    :param sort: Sort DataFrame based on *columns. Default is `True`.
     :param by: label or list of labels to group by.
         The explicit missing rows are returned per group.
+    :param fill_value: Scalar value to use instead of NaN
+        for missing combinations. A dictionary, mapping columns names
+        to a scalar value is also accepted.
+    :param explicit: Determines if only implicitly missing values
+        should be filled (`False`), or all nulls existing in the dataframe
+        (`True`). Default is `True`.
```
Comment (thatlittleboy) [inline janitor/functions/complete.py]: Shall we also include a note here that `explicit` will be ignored if `fill_value` is not provided?

**Review by thatlittleboy — APPROVED** [review]
Just a couple of points, but otherwise the implementation LGTM.
I trust you'll address the comment about the default args, so I'll just pre-approve.

File: janitor/functions/complete.py (hunk 1 above)
Comment (samukweku) [reply janitor/functions/complete.py]: Hmmm ... fair point ... yea, it seems easier on the eyes if the output is sorted. Thinking about it, I will leave the default argument as False, the ease on the eyes can always be achieved by explicitly setting sort = True

File: janitor/functions/complete.py [hunk 3]
```diff
@@ -191,9 +309,7 @@ def _complete_column(column, df):
     A Pandas Series/DataFrame with no duplicates,
     or a list of unique Pandas Series is returned.
     """
-    raise TypeError(
-        """This type is not supported in the `complete` function."""
-    )
+    raise TypeError("This type is not supported in the `complete` function.")
```
Comment (ericmjl) [inline janitor/functions/complete.py]: @samukweku can we get a test that covers this line as well?

**Review by ericmjl — CHANGES_REQUESTED** [review]
Just one change request, @samukweku, to cover a line that isn't currently covered! If not possible, not an issue, I'll pre-approve regardless. @thatlittleboy feel free to merge after @samukweku responds!

**Review by ericmjl — APPROVED** [review]

File: janitor/functions/complete.py (hunk 3 above)
Comment (samukweku) [reply janitor/functions/complete.py]: this check is already covered in the _complete_checks section; it is safe to remove the TypeError and use a default option here.

**Pull request merged**